### PR TITLE
#124 update metadata incrementally

### PR DESCRIPTION
**Pull request by olenagerasimova**
For #124 I've tried implement updating metadata for changed packages only.

### CODE REVIEW

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 11@7d3334767964, thread: PQ-CT2E6TK9B
com.zerocracy.farm.ruled.RdRules[130] java.lang.IllegalStateException: http://datum.zerocracy.com/latest/rules/pm/no-lost-estimates.xsl says: The estimate for job "gh:artipie/maven-adapter#74" exists in estimates.xml, but the job is absent in the WBS among 53 jobs.
```

[0.53.9](https://github.com/zerocracy/farm): CID: [18b59ad6-9e14-454f-baab-4a2695463bfe](https://www.0crat.com/footprint/CT2E6TK9B/18b59ad6-9e14-454f-baab-4a2695463bfe), Type: "Add job to WBS"

File: src/main/java/com/artipie/rpm/ModifiableRepository.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            this.origin.update(pkg);
+        }
+        return this;
+    }
+
+    /**
+     * Deletes extra packages data from metadata.
+     * @return Itself
+     */
+    public ModifiableRepository extract() {
```
[the remark this answers is not in the document]
Comment (olenagerasimova) [reply src/main/java/com/artipie/rpm/ModifiableRepository.java:100]: @g4s8 what verb do you think is better to use here? This method is supposed delete records about unexciting packages from metadata files. Not yet implemented`ModifiableMetadata#delete` method also does not delete any files, is deletes records about unexciting packages from metadata.

[the remark this answers is not in the document]
Comment (g4s8) [reply src/main/java/com/artipie/rpm/ModifiableRepository.java:100]: @olenagerasimova what do you think about "clear" or "prune"?

**Comment by olenagerasimova** [comment]
@g4s8 thanks, I've renamed methods and corrected javadocs.

**Pull request merged**